core/domain/recommendations_services.py

```diff
@@ -390,7 +419,11 @@ def delete_explorations_from_recommendations(exp_ids):
     # objects.
     all_recommending_models = {}
     for exp_id in exp_ids:
-        recommending_models = recs_model_class.query(
+        # Here we used cast because we need to narrow down the return type of
+        # .fetch() method from Sequence to List for more strict typing.
+        recommending_models: (
```

Comment:
Same as above. Lets keep the type of `recommending_models` in the same line 
```py
        recommending_models: Sequence[
            recommendations_models.ExplorationRecommendationsModel
        ] = recs_model_class.query(
            recs_model_class.recommended_exploration_ids == exp_id
        ).fetch()
```

Reply:
Done!